### PR TITLE
add for all dev branch

### PR DESCRIPTION
**Pull request by khanhntd**
**Description:** <Describe what has changed.>

For all of the branch that has the prefix "dev", it would trigger the CI. In this way, we can test the CI.

**Testing:** <Describe what testing was performed and which tests were added.>
An alternative solution would be using act https://github.com/nektos/act

### CODE REVIEW

**Review by sethAmazon — APPROVED** [review]

**Review by jefchien — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ on:
     branches:
       - main
       - release/v*
-      - dev
+      - dev/*
```
Comment (jefchien) [inline .github/workflows/CI.yml:20]: We still have a `dev` branch even if it isn't really in use. Should we still have the CI run on it? This change will remove that.

Comment (khanhntd) [reply .github/workflows/CI.yml:20]: There are two ways for us to change it: one is to rename the branch as dev/ and second is that change the prefix pattern to dev*.

Comment (jefchien) [reply .github/workflows/CI.yml:20]: The other way is to just have two entries
```
- dev
- dev/*
```

Comment (khanhntd) [reply .github/workflows/CI.yml:21]: Reasonable. Fixed it.

**Review by jefchien — APPROVED** [review]

**Pull request merged**